From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: trainb0y <trainb0ydev@gmail.com>
Date: Mon, 10 Apr 2023 16:53:57 -0600
Subject: [PATCH] Allow block entities to be moved


diff --git a/src/main/java/net/minecraft/world/level/block/entity/BlockEntity.java b/src/main/java/net/minecraft/world/level/block/entity/BlockEntity.java
index 370a25d2deb54f10a35ee24d9e7e92fbfde60edf..e24418a75ce01dcaf9a458f7bf91cbe5a2256b69 100644
--- a/src/main/java/net/minecraft/world/level/block/entity/BlockEntity.java
+++ b/src/main/java/net/minecraft/world/level/block/entity/BlockEntity.java
@@ -37,7 +37,7 @@ public abstract class BlockEntity {
     private final BlockEntityType<?> type;
     @Nullable
     protected Level level;
-    protected final BlockPos worldPosition;
+    protected BlockPos worldPosition; // Nebula
     protected boolean remove;
     private BlockState blockState;
 
@@ -180,6 +180,13 @@ public abstract class BlockEntity {
         return this.worldPosition;
     }
 
+    // Nebula start - movable block entities
+    public void setBlockPos(BlockPos pos) {
+        this.worldPosition = pos;
+        setChanged();
+    }
+    // Nebula end
+
     public BlockState getBlockState() {
         return this.blockState;
     }
