From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: trainb0y <trainb0ydev@gmail.com>
Date: Tue, 11 Apr 2023 18:39:40 -0600
Subject: [PATCH] Move Stellarica's setBlockFast to nms Level


diff --git a/src/main/java/net/minecraft/world/level/Level.java b/src/main/java/net/minecraft/world/level/Level.java
index ea8a0961190e9aafda4fed6fecd85097c141040a..dc90c0afbac9585bac2a4bc53901c1ce426bf6ab 100644
--- a/src/main/java/net/minecraft/world/level/Level.java
+++ b/src/main/java/net/minecraft/world/level/Level.java
@@ -1003,6 +1003,28 @@ public abstract class Level implements LevelAccessor, AutoCloseable {
         }
     }
 
+    // Nebula start
+    public void setBlockFast(BlockPos position, BlockState state) {
+        // Distantly derived from the setBlockFast from APDevTeam's NMS code
+        var chunk = getChunkAt(position);
+        var chunkSection = (position.getY() >> 4) - chunk.getMinSection();
+        var section = chunk.getSections()[chunkSection];
+        if (section == null) {
+            // Put a GLASS block to initialize the section. It will be replaced next with the real block.
+            chunk.setBlockState(position, Blocks.GLASS.defaultBlockState(), false);
+            section = chunk.getSections()[chunkSection];
+        }
+        if (section.getBlockState(position.getX() & 15, position.getY() & 15, position.getZ() & 15) == state) {
+            //Block is already of correct type and data, don't overwrite
+            return;
+        }
+        section.setBlockState(position.getX() & 15, position.getY() & 15, position.getZ() & 15, state);
+        sendBlockUpdated(position, state, state, 3);
+        // world.lightEngine.checkBlock(position) // boolean corresponds to if chunk section empty
+        chunk.setUnsaved(true);
+    }
+    // Nebula end
+
     // CraftBukkit start - Split off from above in order to directly send client and physic updates
     public void notifyAndUpdatePhysics(BlockPos blockposition, LevelChunk chunk, BlockState oldBlock, BlockState newBlock, BlockState actualBlock, int i, int j) {
         BlockState iblockdata = newBlock;
