From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: trainb0y <trainb0ydev@gmail.com>
Date: Sun, 16 Apr 2023 09:04:51 -0600
Subject: [PATCH] Disable copper oxidation


diff --git a/src/main/java/net/minecraft/world/level/block/WeatheringCopper.java b/src/main/java/net/minecraft/world/level/block/WeatheringCopper.java
index 8d59c30edfa80ab2c4cfc66b7ba06143afa99e3f..ced6cb93e8067a5fdf27aecd21031e482ae12e96 100644
--- a/src/main/java/net/minecraft/world/level/block/WeatheringCopper.java
+++ b/src/main/java/net/minecraft/world/level/block/WeatheringCopper.java
@@ -47,9 +47,13 @@ public interface WeatheringCopper extends ChangeOverTimeBlock<WeatheringCopper.W
 
     @Override
     default Optional<BlockState> getNext(BlockState state) {
+        /* Nebula start - disable copper oxidation
         return getNext(state.getBlock()).map((block) -> {
             return block.withPropertiesOf(state);
         });
+         */
+        return Optional.of(state);
+        // Nebula end
     }
 
     @Override
